sklearnex/ensemble/_forest.py

```diff
@@ -119,22 +144,49 @@ def _onedal_fit(self, X, y, sample_weight=None, queue=None):
                     sample_weight = sample_weight * expanded_class_weight
                 else:
                     sample_weight = expanded_class_weight
-            if sample_weight is not None:
-                sample_weight = [sample_weight]
+
+            # Decapsulate classes_ attributes following scikit-learn's
+            # BaseForest.fit. oneDAL does not support multi-output, therefore
+            # the logic can be hardcoded in comparison to scikit-learn's logic
+            if hasattr(self, "classes_"):
+                self.n_classes_ = self.n_classes_[0]
+                self.classes_ = self.classes_[0]
+
         else:
             # try catch needed for raw_inputs + array_api data where unlike
             # numpy the way to yield unique values is via `unique_values`
             # This should be removed when refactored for gpu zero-copy
-            try:
-                self.classes_ = xp.unique(y)
-            except AttributeError:
-                self.classes_ = xp.unique_values(y)
-            self.n_classes_ = len(self.classes_)
-        self.n_features_in_ = X.shape[1]
+            if hasattr(self, "class_weight"):
```

Comment:
Can we leverage sklearn's is_classifier functionality here? Assuming the intention for this conditional to identify whether classifier